### PR TITLE
MUC HTTP API

### PR DESCRIPTION
**Pull request by Dzol**
Introduce MVP for the MUC HTTP API:

- [x] Create room
- [x] Invite to room
- [x] Send message to room
- [ ] Refactor & improve source (following a review)
- [x] Improve & expand tests

### CODE REVIEW

File: apps/ejabberd/src/muc_admin.erl [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% Author: Joseph Yiasemides <joseph.yiasemides@erlang-solutions.com>
+%% Description: Administration commands for Mult-user Chat (MUC)
+%%==============================================================================
+
+-module(muc_admin).
```
Comment (michalwski) [inline apps/ejabberd/src/muc_admin.erl]: All modules implementing `gen_mod` behaviour are named with `mod_` prefix. This one should also follow this convention.

File: apps/ejabberd/src/muc_admin.erl [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+start(_, _) ->
+    mongoose_commands:register(commands()).
+
+stop(_) ->
+    mongoose_commands:unregister(commands()).
+
+commands() ->
+    [
+     [{name, create_muc_room},
+      {category, muc},
```
Comment (michalwski) [inline apps/ejabberd/src/muc_admin.erl]: The category is the later translated to the resource in REST path. The good practise is to use plural nouns for that. I think category name `mucs` will suite better. But I'm open to here a constructive disagreement :)

File: apps/ejabberd/src/muc_admin.erl [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      {desc, "Create a MUC room."},
+      {module, ?MODULE},
+      {function, create_room},
+      {action, create},
+      {identifiers, [domain]},
```
Comment (michalwski) [inline apps/ejabberd/src/muc_admin.erl]: I think better name would be `host` instead of `domain`.

File: apps/ejabberd/src/muc_admin.erl [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      {args,
+       %% The argument `domain' is what we normally term the XMPP
+       %% host, `name' is the room name, `owner' is the XMPP entity
+       %% that would normally request an instant MUC room.
+       [{domain, binary},
+        {name, binary},
+        {owner, binary},
```
Comment (michalwski) [inline apps/ejabberd/src/muc_admin.erl]: Can we somehow sidestep the owner parameter? By f.e reading the owner?

Comment (michalwski) [inline apps/ejabberd/src/muc_admin.erl]: Sorry, I forgot it's create method. Forget my previous comment.

**Comment by Dzol** [comment]
@michalwski: Commit a233e60 should resolve the comments above.

**Pull request closed without merging**